cmd/yggd/client.go

```diff
@@ -11,41 +11,123 @@ import (
 	"git.sr.ht/~spc/go-log"
 	"github.com/google/uuid"
 	"github.com/redhatinsights/yggdrasil"
+	"github.com/redhatinsights/yggdrasil/internal/config"
 	"github.com/redhatinsights/yggdrasil/internal/transport"
+	"github.com/redhatinsights/yggdrasil/internal/work"
 )
 
 type Client struct {
 	t transport.Transporter
-	d *dispatcher
+	d *work.Dispatcher
 }
 
+// NewClient creates a new Client configured with dispatcher and transporter.
+func NewClient(dispatcher *work.Dispatcher, transporter transport.Transporter) *Client {
+	return &Client{
+		d: dispatcher,
+		t: transporter,
```

Comment:
TBH, I would change t to transporter, so reading the rest of the project is a bit easier

Reply:
Good point. I renamed both `t` and `d` to the full word.